CryptoPrices/Sources/Data/Sources/NetworkCore/Models/DataPoint.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  DataPoint.swift
+//  Data
+//
+//  Created by Minh Pham on 27/12/2022.
+//
+
+import Foundation
+
+public struct DataPoint {
+
+    public let timestamp: TimeInterval
+    public let price: Double
+
+    public init(timestamp: TimeInterval, price: Double) {
+        self.timestamp = timestamp
+        self.price = price
+    }
+}
```

Comment:
@Thieurom this model is not a direct map of the API's response (the response doesn't contain any fields that have `timestamp` and  `price`), so that is why I intentionally remove the word `API` from the model 😅 And thus, I don't think this class can be replaced with `APIPrice` from your suggestion, due to another reason that is `Price` is not the right naming for a model that contains timestamp and price, logically it is a chart data point 🙏

Reply:
Yeah, I got your point here 🙏

But as you mentioned,
>the `DataPoint` is not a direct map of the API's response

and 
> shouldn't we return the exact response object and the mapping should be done via the Repository instead

why bother putting it in the network layer? Why don't put this `DataPoint` directly in the `Repositories` module? I tried and worked fine.

<img width="275" alt="Screenshot 2022-12-29 at 11 25 31" src="https://user-images.githubusercontent.com/17972674/209903312-76f82bf4-fdc4-42e9-9d71-6da2506bbffa.png">

That is because `DataPoint` is a kind of intermediate type for transforming data purposes, used specifically on the implementation detail of the `Repositories`. At the end of the day, what it returns is still the abstract type (`CharDataPoint` protocol) by casting. So the dependee (`GetChartPricesUseCase`) doesn't know that. You can omit the `public` marks on the definition of `DataPoint` too to make it internally to `Repositories`.

Reply:
Good point @Thieurom, let me omit the public marks on the definition of DataPoint too to make it internally to Repositories. 👍

Reply:
> At the end of the day, what it returns is still the abstract type (CharDataPoint protocol) by casting. So the dependee (GetChartPricesUseCase) doesn't know that. You can omit the public marks on the definition of DataPoint too to make it internally to Repositories.

-> Even though `DataPoint` is a kind of intermediate type for transforming data purposes used specifically on the implementation detail of the `Repositories`, I still think it is better to place it in `NetworkCore` instead of `Repositories`. This is because:
- The current implementation for `DataPoint` is just like what we are doing with the API models like `APICoin` and `APIPrices` as the UseCase also doesn't know about them, but we still make them public for `NetworkExtension`, `DataTestHelpers`, etc. to use, right? Isn't `NetworkCore` the place we should define the **models** for even internal libs of the Data layer to use? 🤔
- It just feels weird that we treat the `DataPoint` class uniquely in a way that `Repositories` can contain other types of classes than repository implementation classes just to limit the access of the `DataPoint` 😅

Reply:
- We don't put the `DataPoint` in the network layer because both interface (`NetworkCore`) and implementation (`NetworkExtension`) don't use it. Its functionality returns `APICoin` and `APIPrices`, which it already includes. But yeah, that's usually fine to define some extra, convenient types for the module consumer 🙏 
- >It just feels weird that we treat the DataPoint class uniquely in a way that Repositories can contain other types of classes than repository implementation classes just to limit the access of the DataPoint

  I don't quite understand your point here 🙏 It means you can't add any new types in the `Repositories` module to support its implementation of the `CoinRepositoryProtocol`? Why is that 🤔 Let's take the `Home` module as an example:
The `Home` module makes a type `HomeView` publicly which conforms to the SwiftUI's `View` protocol. It means it has to have a `body` property with the type of `some View`. In doing so, the `Home` module defines some intermediate types, but for internal use without public them like `WalletStatisticSection`, `TrendingCoinSection`, ... which also just conforms to the `View` protocol. In the end, what `body` returns is `some View` so the consumer doesn't know and doesn't care what exact types it receives under the hood. This is precisely the same situation with the `DataPoint` that conforms to `ChartDataPoint`.

But to conclude, I think putting it in the network layer doesn't hurt, and it has some potential benefits as you mentioned. So let's keep it as is 👍

Reply:
> I don't quite understand your point here 🙏 It means you can't add any new types in the Repositories module to support its implementation of the CoinRepositoryProtocol? Why is that 🤔

-> Ah FYI originally, it is because the name of the library here is `Repositories`, so from my impression, we should only put Repository implementation classes here only, and no model classes like `DataPoint` should be included 🙏. For example:
- `CoinRepository`
- `UserRepository` (for authentication feature)
- `PaymentRepository` (for payment feature)
- So on...

Hope this helps clarify the confusion! 🙏 I see that you have the Home module example there but I think it is a bit off-track of my point 🙈

Reply:
As an updated note, we can still keep the `DataPoint` in the network layer and make it public as is, but we shouldn't attempt to use it on the use case or any higher layer because
>A module of a lower layer depends on one or more modules of the higher layer, but not vice versa

Reply:
Yup, I agree with that idea too 👍